python/lsst/cp/verify/report.py

```diff
@@ -0,0 +1,780 @@
+                    # Is a results catalog
+                    self.block(ds, self.out_files["index.html"], ds['uri'])
+                    self.include(ds['uri'].path, "index.html")
+                elif ds['exposure'] is None and ds['detector'] is None:
+                    self.block(ds, self.out_files["index.html"])
+                    if ".png" in ds['uri'].path:
+                        self.image_handler(ds['uri'].path, ds['instrument'], self.out_files["index.html"])
+                    else:
+                        self.include(ds['uri'].path, "index.html")
+
+        # Record all datasets here, and link to them directly:
+        page = self.out_files["manifest.html"]
+        page.extend(["<table width='100%'>",
+                     "<tr>",
+                     "<th>stage</th>", "<th>ref</th>", "<th>type</th>", "<th>storage_class</th>",
+                     "<th>dataId</th>", "<th>collection</th>",
+                     "</tr>"])
+        for ds in self.datasets:
+            relative_file = self.relative_file(ds['uri'].path)
+            page.extend(["<tr>",
+                         f"<td>{ds['stage']}</td>", f"<td>{ds['ref']}</td>",
+                         "<td>", f'<a href="{relative_file}">', f"{ds['type']}", "</a>",
+                         f"<td>{ds['storage_class']}</td>",
+                         f"<td>{ds['dataId']}</td>", f"<td>{ds['collection']}</td>",
+                         "</tr>"])
+
+    # File conversion utilities:
+    @staticmethod
+    def fits_to_png(data, target_uri, title):
+        """Convert FITS to PNG, using the same scaling as on RubinTV.
+
+        Parameters
+        ----------
+        data : `lsst.afw.Image`, `lsst.afw.MaskedImage` or `lsst.afw.Exposure`
+            The fits image data to convert.
+        target_uri : `str`
+            Path to the PNG file to write.
+        title : `str`
+            Title to add to the figure.
+
+        See Also
+        --------
+        rubintv_production / mosaicing.py
+        """
+        # Get array from either an Image or an Exposure:
+        try:
+            array = data.image.array
+        except AttributeError:
+            array = data.array
+
+        fig = make_figure()
+        ax = fig.gca()
+        ax.clear()
+        cmap = cm.gray
+        # This was using summit_utils.getQuantiles, but that
+        # was blowing out the scaling more than I wanted.
+        q25, q50, q75 = np.nanpercentile(array, [25, 50, 75])
+        scale = 3.0 * 0.74 * (q75 - q25)
+        quantiles = np.arange(q50 - scale, q50 + scale, 2.0 * scale / cmap.N)
+        norm = colors.BoundaryNorm(quantiles, cmap.N)
+
+        im = ax.imshow(array, norm=norm, interpolation='None', cmap=cmap, origin='lower')
+
+        divider = make_axes_locatable(ax)
+        cax = divider.append_axes("right", size="5%", pad=0.05)
+        fig.suptitle(title)
+        fig.colorbar(im, cax=cax)
+        fig.tight_layout()
+        fig.savefig(target_uri.path)
```

Comment:
Do you want a plt.close here?

Reply:
I don't know.  If I'm reading the changelog for `make_figure` correctly (https://github.com/lsst/utils/blob/7d0def3d61c752f736c8d2cc33852495fcc90713/doc/changes/DM-44725.md?plain=1), then I think no, because this figure is independent of `pyplot` in some fashion.  I've not added one, and hope that when the method ends, the `fig` drops out of scope and is cleared.